.github/actions/build/action.yml

```diff
@@ -0,0 +1,35 @@
+name: Setup
+description: "Setup Project"
+runs:
+  using: "composite"
+  steps:
+    - name: Get yarn cache directory path
+      id: yarn-cache-dir-path
+      run: echo "::set-output name=dir::$(yarn cache dir)"
+      shell: bash
+
+    - uses: actions/cache@v3
+      id: yarn-cache
+      with:
+        path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+        key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-yarn-
```

Comment:
is the `yarn-` missing a suffic?

Reply:
Nope see below, it is meant to be a prefix. This will pull caches event if they don't match exactly since there is a high likelihood many of the dependencies will be the same.